ra2ce/analysis/damages/damage_calculation/damage_network_return_periods.py

```diff
@@ -93,99 +96,107 @@ def main(self, damage_function: DamageCurveEnum, manual_damage_functions):
-                # _max_RP = max(_to_integrate.columns)
-
-            self.gdf["risk"] = _risk
-
-        elif mode == RiskCalculationModeEnum.TRIANGLE_TO_NULL_YEAR:
-            """
-            In this mode, an extra data point with zero damage is added at some distance from the smallest known RP,
-            and the area of the Triangle this creates is also calculated
-            """
-
-            _rps = list(_to_integrate.columns)
-
-            if year >= min(_rps):
-                raise ValueError(
-                    """
-                RA2CE cannot calculate risk in 'triangle_to_null' mode if 
-                Return period of the triangle ({}) >= smallest available return period ({})
-                Use 'default' mode or 'cut_from' instead.
-                                    """.format(
-                        year, min(_rps)
+            elif mode == RiskCalculationModeEnum.CUT_FROM_YEAR:
+                """
+                In this mode, the integration mimics the presence of a flood protection
+                """
+                _rps = list(_to_integrate.columns)
+
+                if year <= min(_rps):
+                    raise ValueError(
+                        """
+                    RA2CE cannot calculate risk in 'cut_from' mode if 
+                    Return period of the cutoff ({}) <= smallest available return period ({})
+                    Use 'default' mode or 'triangle_to_null_mode' instead.
+                                        """.format(
+                            year, min(_rps)
+                        )
                     )
-                )
 
-            _to_integrate = self.rework_damage_data_triangle_to_null(
-                _to_integrate, year
-            )
-            _risk = self.integrate_df_trapezoidal(_to_integrate.copy())
+                elif (
+                    min(_rps) < year < max(_rps)
+                ):  # if protection level is between min and max RP
+                    _to_integrate = self.rework_damage_data_cut_from(
+                        _to_integrate, year
+                    )
+                    _risk = self.integrate_df_trapezoidal(_to_integrate.copy())
+
+                elif year >= max(
+                    _rps
+                ):  # cutoff is larger or equal than largest return period
+                    # risk is return frequency of cutoff
+                    # times the damage of the most extreme event
+                    _to_integrate = _to_integrate.fillna(0)
+                    _risk = _to_integrate[max(_to_integrate.columns)] / year
+
+            elif mode == RiskCalculationModeEnum.TRIANGLE_TO_NULL_YEAR:
+                """
+                In this mode, an extra data point with zero damage is added at some distance from the smallest known RP,
+                and the area of the Triangle this creates is also calculated
+                """
+
+                _rps = list(_to_integrate.columns)
```

Comment:
looks like_rps can be declared before the `if` statement on mode?

Reply:
Yes, but mode can have only one value => _rps can be defined once

Did I understood what you meant correctly?

Reply:
i meant that the line
`_rps = list(_to_integrate.columns)` 
is repeated several (maybe unnecessarily)

Reply:
Ow alright! revised.